### PR TITLE
Ext 231

### PR DESCRIPTION
**Pull request by Lestaf-Pimentel**
Solved EXT-231 passing Bluecheck's configuration to onShippingSuccess, which contains the shipping fields as arguments, during debugging also soved 244.

In regards to 244, added a check on component update which analyzed the instance's current state with the past one and reloaded the script when going from billing to shipping step.

There was also a bug when reloading the page that caused componintDidUnmount to attempt deletion when the script was not yet present, displaying an error message, so that was fixed as well.

### CODE REVIEW

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 1]
```diff
@@ -11,10 +11,13 @@
 
 import { isSignedIn } from 'Util/Auth';
 import BrowserDatabase from 'Util/BrowserDatabase';
+//import { connect } from 'react-redux';
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove commented code

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 2]
```diff
@@ -11,10 +11,13 @@
...
 
-import blueCheckScriptEnabler from '../../../static/blueCheckEnablerScript';
+//import blueCheckScriptEnabler from '../../../static/blueCheckEnablerScript';
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove commented code

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 3]
```diff
@@ -30,34 +33,142 @@ export const loadScript = () => {
...
-export const configBlueCheck = () => {
-    const isCustomerSignedIn = isSignedIn();
-    const customer = BrowserDatabase.getItem('customer') || {};
-    return blueCheckScriptEnabler(isCustomerSignedIn, customer.addresses, customer.email);
+export const scrapeLoggedInUserData = (addresses, email) => {
+    const chosenAddressId = Number(
+        document.getElementsByClassName('CheckoutAddressTable-Button_isSelected')[0]
+            .parentElement.classList[1]
+    );
+    console.log("WHAEIHIFqmNAK")
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove console.logs

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 4]
```diff
@@ -30,34 +33,142 @@ export const loadScript = () => {
...
 };
 
-export const componentDidMount = (args, callback, _instance) => {
-    loadScript().then(() => {
-        configBlueCheck();
-    });
 
+export const scrapeGuestUserData = (email = null,address) => {
+    const userDataPrefix = 'shipping';
+    console.log('AAAAAAAAAA')
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove console.logs

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 5]
```diff
@@ -30,34 +33,142 @@ export const loadScript = () => {
...
+    BlueCheck.userData[`${userDataPrefix }_address`] = address.street0;
+    BlueCheck.userData[`${userDataPrefix }_address2`] = address.street1;
+    BlueCheck.userData[`${userDataPrefix }_address3`] =address.street2;
+    BlueCheck.userData[`${userDataPrefix }_city`] = address.city;
+    BlueCheck.userData[`${userDataPrefix }_country`] = address.country_id;
+    BlueCheck.userData[`${userDataPrefix }_region`] = address.region_id;
+    BlueCheck.userData[`${userDataPrefix }_phone`] = address.telephone;
+    BlueCheck.userData[`${userDataPrefix }_postal_code`] =address.postcode;
+    BlueCheck.requiredFields = [`${userDataPrefix }_first_name`,`${userDataPrefix }_last_name`,`${userDataPrefix }_country`, 'email'];
+    console.log(BlueCheck)
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove console.logs

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 6]
```diff
@@ -30,34 +33,142 @@ export const loadScript = () => {
...
+        BlueCheck.platformCallbacks.onReady = () => {
+            function prepareButton() {
+                const verifyBtn = document.getElementsByClassName('CheckoutShipping-Button')[0];
+                console.log(verifyBtn);
+                    BlueCheck.platformCallbacks.onSuccess = () => {
+                        BrowserDatabase.setItem(true,'blueCheck')
+                        verifyBtn.click();
+                    };
+                    if (BlueCheck.validateAndDisplayModal()) {
+                        //e.stopPropagation();
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove commented code

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 7]
```diff
@@ -30,34 +33,142 @@ export const loadScript = () => {
...
+export const onShippingSuccess = (args,callback,_instance) => {
+    const isCustomerSignedIn = isSignedIn();
+    const customer = BrowserDatabase.getItem('customer') || {};
+    const { BlueCheck } = window;
+    
+    if(BlueCheck){
+        BlueCheck.platformCallbacks.onReady = () => {
+            function prepareButton() {
+                const verifyBtn = document.getElementsByClassName('CheckoutShipping-Button')[0];
+                console.log(verifyBtn);
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: remove console.logs

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 8]
```diff
@@ -1,31 +1,38 @@
+import BrowserDatabase from 'Util/BrowserDatabase';
 export const TIMEOUT_LISTEN_TO_CHECKOUT = 250;
 
+//keeping this for posterity
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove this lines

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 9]
```diff
@@ -1,31 +1,38 @@
...
-    BlueCheck.userData[`${userDataPrefix }_last_name`] = document.getElementById('lastname').value;
-    BlueCheck.userData[`${userDataPrefix }_address`] = document.getElementById('street0').value;
-    BlueCheck.userData[`${userDataPrefix }_address2`] = document.getElementById('street1').value;
-    BlueCheck.userData[`${userDataPrefix }_address3`] = document.getElementById('street2').value;
-    BlueCheck.userData[`${userDataPrefix }_city`] = document.getElementById('city').value;
-    const country = document.getElementById('country_id');
-    if (country) {
-        BlueCheck.userData[`${userDataPrefix }_country`] = country.options[country.options.selectedIndex].innerText;
-    }
+    console.log('scraping...', email);
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove console log

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 10]
```diff
@@ -1,31 +1,38 @@
...
+    BlueCheck.userData.email = email || "II@IIII.II"//ithisstate.state.shippingAddress.email;
+    BlueCheck.userData[`${userDataPrefix }_first_name`] = 1//thisstate.state.shippingAddress.firstname;
+    BlueCheck.userData[`${userDataPrefix }_last_name`] = 1//thisstate.state.shippingAddress.lastname;
+    BlueCheck.userData[`${userDataPrefix }_address`] = 1//thisstate.state.shippingAddress.street[0];
+    BlueCheck.userData[`${userDataPrefix }_address2`] = 1//thisstate.state.shippingAddress.street[1];
+    BlueCheck.userData[`${userDataPrefix }_address3`] =1// thisstate.state.shippingAddress.street[2];
+    BlueCheck.userData[`${userDataPrefix }_city`] = 1//thisstate.state.shippingAddress.city;
+    BlueCheck.userData[`${userDataPrefix }_country`] = 1//thisstate.state.shippingAddress.country_id;
+    //const country = document.getElementById('country_id');
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove commented code

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 11]
```diff
@@ -1,31 +1,38 @@
...
+    //}
 
-    const region = document.getElementById('region_id');
-    if (region) {
-        BlueCheck.userData[`${userDataPrefix }_region`] = region.options[region.options.selectedIndex].innerText;
-    }
-    BlueCheck.userData[`${userDataPrefix }_phone`] = document.getElementById('telephone').value;
-    BlueCheck.userData[`${userDataPrefix }_postal_code`] = document.getElementById('postcode').value;
+    BlueCheck.userData[`${userDataPrefix }_region`] = 1//state.region_id;
+    //const region = document.getElementById('region_id');
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove comented code

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 12]
```diff
@@ -1,31 +1,38 @@
...
+    //if (region) {
+    //    BlueCheck.userData[`${userDataPrefix }_region`] = region.options[region.options.selectedIndex].innerText;
+    //}
+    BlueCheck.userData[`${userDataPrefix }_phone`] = 1//state.telephone;
+    BlueCheck.userData[`${userDataPrefix }_postal_code`] =1// state.postcode;
     BlueCheck.requiredFields = [`
     ${userDataPrefix }_first_name`, `${userDataPrefix }_last_name`,
     `${userDataPrefix }_country`, 'email'];
+    console.log(BlueCheck.userData);
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove log

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 13]
```diff
@@ -55,13 +62,16 @@ export const scrapeLoggedInUserData = (addresses, email) => {
     ];
 };
 
-export default (isCustomerSignedIn, addresses, email) => {
+export default (isCustomerSignedIn, addresses, email, state) => {
     const { BlueCheck } = window;
+    console.log('starting...');
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove log

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 14]
```diff
@@ -55,13 +62,16 @@ export const scrapeLoggedInUserData = (addresses, email) => {
...
     BlueCheck.platformCallbacks.onReady = () => {
         function prepareButton() {
             const verifyBtn = document.getElementsByClassName('CheckoutShipping-Button')[0];
+            console.log(verifyBtn);
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove log

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 15]
```diff
@@ -71,6 +81,7 @@ export default (isCustomerSignedIn, addresses, email) => {
             };
         }
         function listenForCheckout() {
+            console.log('listening forcheckout')
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove loging

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 16]
```diff
@@ -80,11 +91,14 @@ export default (isCustomerSignedIn, addresses, email) => {
         setTimeout(listenForCheckout);
     };
     BlueCheck.platformCallbacks.scrapeUserData = () => {
+        console.log('StartedScarping')
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove loging

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 17]
```diff
@@ -80,11 +91,14 @@ export default (isCustomerSignedIn, addresses, email) => {
...
         const isCustomAddress = document
             .getElementsByClassName('CheckoutAddressBook-Button_isCustomAddressExpanded')[0] !== undefined;
 
         return (isCustomerSignedIn && !isCustomAddress)
-            ? scrapeLoggedInUserData(addresses, email) : scrapeGuestUserData(email);
+            ? /*scrapeLoggedInUserData(addresses, email)*/console.log("weeee") : console.log("WAAAA")//scrapeGuestUserData(email);
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove comented code

File: pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js [hunk 18]
```diff
@@ -80,11 +91,14 @@ export default (isCustomerSignedIn, addresses, email) => {
...
     };
+    
     BlueCheck.initialize();
+    console.log("O O BUBU",BlueCheck)
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/static/blueCheckEnablerScript.js]: remove loging

File: pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js [hunk 19]
```diff
@@ -30,34 +33,142 @@ export const loadScript = () => {
...
+    BlueCheck.userData[`${userDataPrefix }_city`] = address.city;
+    BlueCheck.userData[`${userDataPrefix }_country`] = address.country_id;
+    BlueCheck.userData[`${userDataPrefix }_region`] = address.region_id;
+    BlueCheck.userData[`${userDataPrefix }_phone`] = address.telephone;
+    BlueCheck.userData[`${userDataPrefix }_postal_code`] =address.postcode;
+    BlueCheck.requiredFields = [`${userDataPrefix }_first_name`,`${userDataPrefix }_last_name`,`${userDataPrefix }_country`, 'email'];
+    console.log(BlueCheck)
+};
+
+
```
Comment (GAkim) [inline pwa/packages/bluecheckageverify-spwa/src/plugin/route/Checkout/CheckoutContainer.plugin.js]: no need in to blank lines

**Review by GAkim — CHANGES_REQUESTED** [review]

**Comment by Lestaf-Pimentel** [comment]
Fixed formatting problems, deleted redundant file

**Review by GAkim — APPROVED** [review]

**Pull request merged**